src/main/codec.ts

```diff
@@ -1,23 +1,26 @@
-import { BinaryProtocol, BufferedTransport, StructLike, TProtocol } from '@creditkarma/thrift-server-core'
+import { BinaryProtocol, BufferedTransport,
+    CompactProtocol, StructLike, TProtocol } from '@creditkarma/thrift-server-core'
 
 export interface IStructConstructor<T extends StructLike> {
     new (args?: any): T
     read(input: TProtocol): T
 }
 
-export const encoder = (thriftObject: StructLike): Promise<Buffer> => {
+export const encoder = (thriftObject: StructLike, useCompactProtocol: boolean = false): Promise<Buffer> => {
```

Comment:
Talked to you about this in person, just putting here for reference. I'd like to see `useCompactProtocol` actually be replace with the protocol class.

```
export const encoder = (thriftObject: StructLike, protocol: IProtocolConstructor = BinaryProtocol): Promise<Buffer>
```

Also of note, I believe that `thrift-server-core` exports an interface for `IProtocolConstructor`.